src/bootstrap/test.rs

```diff
@@ -938,8 +947,11 @@ impl Step for RustdocGUI {
             .arg("--tests-folder")
             .arg(builder.build.src.join("src/test/rustdoc-gui"));
         for path in &builder.paths {
-            if let Some(name) = path.file_name().and_then(|f| f.to_str()) {
-                if name.ends_with(".goml") {
+            if let Some(p) = util::is_valid_test_suite_arg(path, "src/test/rustdoc-gui", builder) {
+                if !p.ends_with(".goml") {
+                    continue;
```

Comment:
Why this filter?

I think we should give an error message if someone passes a non-goml path inside src/test/rustdoc-gui. (And similar for the other cases where you added a similar filter).

Reply:
Fine by me!